test/gpu/octree/test_approx_nearest.cpp

```diff
@@ -112,13 +114,9 @@ TEST(PCL_OctreeGPU, approxNearesSearch)
 
   ASSERT_EQ(downloaded, result_host_gpu);
 
-  // find inconsistencies with gpu and cpu cuda impementation
-  // int count_gpu_better = 0;
-  // int count_pcl_better = 0;
-  // int count_different = 0;
   for (size_t i = 0; i < queries.size(); ++i) {
     ASSERT_EQ(dists_pcl[i], dists_gpu[i]);
-    ASSERT_EQ(dists_pcl[i], dists_gpu[i]);
+    ASSERT_NEAR(dists_gpu[i], dists_device_downloaded[i], 0.001);
```

Comment:
In this test you're only comparing results between two implementations.

There should be also a check comparing against and actual numerical distance value since you are able to predict what is the closest point to your query in advance.